sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/BatchParserSuite.scala

```diff
@@ -0,0 +1,155 @@
+    assert(tree.collection.forall(_.isInstanceOf[SparkStatementWithPlan]))
+
+    batch.split(";")
+      .map(cleanupStatementString)
+      .zip(tree.collection)
+      .foreach { case (expected, statement) =>
+        val sparkStatement = statement.asInstanceOf[SparkStatementWithPlan]
+        val statementText = sparkStatement.getText(batch)
+        assert(statementText == expected)
+      }
+  }
+
+  test("multi statement") {
+    val batch =
+      """
+        |BEGIN
+        |  SELECT 1;
+        |  SELECT 2;
+        |  INSERT INTO A VALUES (a, b, 3);
+        |  SELECT a, b, c FROM T;
+        |  SELECT * FROM T;
+        |END""".stripMargin
+    val tree = parseBatch(batch)
+    assert(tree.collection.length == 5)
+    assert(tree.collection.forall(_.isInstanceOf[SparkStatementWithPlan]))
+    batch.split(";")
+      .map(cleanupStatementString)
+      .zip(tree.collection)
+      .foreach { case (expected, statement) =>
+        val sparkStatement = statement.asInstanceOf[SparkStatementWithPlan]
+        val statementText = sparkStatement.getText(batch)
+        assert(statementText == expected)
+      }
+  }
+
+  test("multi statement without ; at the end") {
+    val batch =
+      """
+        |BEGIN
+        |SELECT 1;
+        |SELECT 2;
+        |INSERT INTO A VALUES (a, b, 3);
+        |SELECT a, b, c FROM T;
+        |SELECT * FROM T
+        |END""".stripMargin
+    val tree = parseBatch(batch)
+    assert(tree.collection.length == 5)
+    assert(tree.collection.forall(_.isInstanceOf[SparkStatementWithPlan]))
+    batch.split(";")
+      .map(cleanupStatementString)
+      .zip(tree.collection)
+      .foreach { case (expected, statement) =>
+        val sparkStatement = statement.asInstanceOf[SparkStatementWithPlan]
+        val statementText = sparkStatement.getText(batch)
+        assert(statementText == expected)
+      }
+  }
+
+  test("nested begin end") {
+    val batch =
+      """
+        |BEGIN
+        |  BEGIN
+        |  SELECT 1;
+        |  END;
+        |  BEGIN
+        |    BEGIN
+        |      SELECT 2;
+        |      SELECT 3;
+        |    END;
+        |  END;
+        |END""".stripMargin
+    val tree = parseBatch(batch)
+    assert(tree.collection.length == 2)
+    assert(tree.collection.head.isInstanceOf[BatchBody])
+    val body1 = tree.collection.head.asInstanceOf[BatchBody]
+    assert(body1.collection.length == 1)
+    assert(body1.collection.head.asInstanceOf[SparkStatementWithPlan].getText(batch) == "SELECT 1")
+
+    val body2 = tree.collection(1).asInstanceOf[BatchBody]
+    assert(body2.collection.length == 1)
+    assert(body2.collection.head.isInstanceOf[BatchBody])
+    val nestedBody = body2.collection.head.asInstanceOf[BatchBody]
+    assert(
+      nestedBody.collection.head.asInstanceOf[SparkStatementWithPlan].getText(batch) == "SELECT 2")
+    assert(
+      nestedBody.collection(1).asInstanceOf[SparkStatementWithPlan].getText(batch) == "SELECT 3")
+  }
+
+  // Helper methods
```

Comment:
Is there any way in Scala to add section comments?
Or any other way to achieve anything similar?

Reply:
Seems like there isn't => hackathon project idea maybe...